jerry-core/ecma/builtin-objects/ecma-builtin-global.cpp

```diff
@@ -1329,6 +1329,122 @@ ecma_builtin_global_object_escape (ecma_value_t this_arg __attr_unused___, /**<
   return ret_value;
 } /* ecma_builtin_global_object_escape */
 
+/**
+ * The Global object's 'unescape' routine
+ *
+ * See also:
+ *          ECMA-262 v5, B.2.2
+ *
+ * @return completion value
+ *         Returned value must be freed with ecma_free_completion_value.
+ */
+static ecma_completion_value_t
+ecma_builtin_global_object_unescape (ecma_value_t this_arg __attr_unused___, /**< this argument */
+                                     ecma_value_t arg) /**< routine's first argument */
+{
+  ecma_completion_value_t ret_value = ecma_make_empty_completion_value ();
+
+  /* 1. */
+  ECMA_TRY_CATCH (string, ecma_op_to_string (arg), ret_value);
+  ecma_string_t *input_string_p = ecma_get_string_from_value (string);
+  /* 2. */
+  lit_utf8_size_t input_size = ecma_string_get_size (input_string_p);
+
+  /* 3. */
+  MEM_DEFINE_LOCAL_ARRAY (input_start_p, input_size, lit_utf8_byte_t);
+  ssize_t sz = ecma_string_to_utf8_string (input_string_p, input_start_p, (ssize_t) (input_size));
+  JERRY_ASSERT (sz >= 0);
+
+  lit_utf8_iterator_t iterator = lit_utf8_iterator_create (input_start_p, input_size);
+  /* 4. */
+  /* The length of input string is always greater than output string
+   * so we re-use the input string buffer.
+   * E.g. %xx is 3 byte long, and the maximum is 0xff, which encoded
+   * as 2 bytes in UTF8. Etc. */
```

Comment:
I just suggested an text idea. I think this would sound better:

The %xx is three byte long, and the maximum encoded value is 0xff, which maximum encoded length is two byte. Similar to this, the maximum encoded length of %uxxxx is four byte.